library/src/main/java/com/novoda/downloadmanager/lib/IOHelpers.java

```diff
@@ -0,0 +1,46 @@
+package com.novoda.downloadmanager.lib;
+
+import com.novoda.notils.logger.simple.Log;
+
+import java.io.Closeable;
+import java.io.FileDescriptor;
+import java.io.IOException;
+import java.io.OutputStream;
+
+final class IOHelpers {
+
+    private IOHelpers() {
+    }
+
+    /**
+     * Closes an outputStream and the FileDescriptor associated with it after ensuring all data as been written to disk.
+     */
+    static void closeAfterWrite(OutputStream out, FileDescriptor outFd) {
+        try {
+            if (out != null) {
+                out.flush();
+            }
+            if (outFd != null) {
+                outFd.sync();
+            }
+        } catch (IOException e) {
+            Log.e("Fail sync");
+        } finally {
+            closeQuietly(out);
+        }
+    }
+
+    /**
+     * Closes a resource ignoring any error in the process.
+     */
+    static void closeQuietly(Closeable closeable) {
```

Comment:
http://stackoverflow.com/questions/13878908/sqlitedatabase-does-not-implement-interface

> android.database.sqlite.SQLiteClosable implements java.io.Closeable only from API Level 16.